builtin/providers/aws/resource_aws_api_gateway_api_key.go

```diff
@@ -0,0 +1,226 @@
+	log.Printf("[DEBUG] Reading API Gateway API Key: %s", d.Id())
+
+	apiKey, err := conn.GetApiKey(&apigateway.GetApiKeyInput{
+		ApiKey: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", *apiKey.Name)
+	d.Set("description", *apiKey.Description)
+	d.Set("enabled", *apiKey.Enabled)
+
+	return nil
+}
+
+func resourceAwsApiGatewayApiKeyUpdateOperations(d *schema.ResourceData) []*apigateway.PatchOperation {
+	operations := make([]*apigateway.PatchOperation, 0)
+	if d.HasChange("enabled") {
+		isEnabled := "false"
+		if d.Get("enabled").(bool) {
+			isEnabled = "true"
+		}
+		operations = append(operations, &apigateway.PatchOperation{
+			Op:    aws.String("replace"),
+			Path:  aws.String("/enabled"),
+			Value: aws.String(isEnabled),
+		})
+	}
+
+	if d.HasChange("description") {
+		operations = append(operations, &apigateway.PatchOperation{
+			Op:    aws.String("replace"),
+			Path:  aws.String("/description"),
+			Value: aws.String(d.Get("description").(string)),
+		})
+	}
+
+	if d.HasChange("stage_key") {
+		prev, curr := d.GetChange("stage_key")
+		prevList := prev.(*schema.Set).List()
+		currList := curr.(*schema.Set).List()
+
+		for i := range prevList {
+			p := prevList[i].(map[string]interface{})
+			exists := false
+
+			for j := range currList {
+				c := currList[j].(map[string]interface{})
+				if c["api_id"].(string) == p["api_id"].(string) && c["stage_name"].(string) == p["stage_name"].(string) {
+					exists = true
+				}
+			}
+
+			if !exists {
+				operations = append(operations, &apigateway.PatchOperation{
+					Op:    aws.String("remove"),
+					Path:  aws.String("/stages"),
+					Value: aws.String(fmt.Sprintf("%s/%s", p["api_id"].(string), p["stage_name"].(string))),
+				})
+			}
+		}
+
+		for i := range currList {
+			c := currList[i].(map[string]interface{})
+			exists := false
+
+			for j := range prevList {
+				p := prevList[j].(map[string]interface{})
+				if c["api_id"].(string) == p["api_id"].(string) && c["stage_name"].(string) == p["stage_name"].(string) {
+					exists = true
+				}
+			}
+
+			if !exists {
+				operations = append(operations, &apigateway.PatchOperation{
+					Op:    aws.String("add"),
+					Path:  aws.String("/stages"),
+					Value: aws.String(fmt.Sprintf("%s/%s", c["api_id"].(string), c["stage_name"].(string))),
+				})
+			}
+		}
+	}
+	return operations
+}
+
+func resourceAwsApiGatewayApiKeyUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[DEBUG] Updating API Gateway API Key: %s", d.Id())
+
+	apiKey, err := conn.UpdateApiKey(&apigateway.UpdateApiKeyInput{
+		ApiKey:          aws.String(d.Id()),
+		PatchOperations: resourceAwsApiGatewayApiKeyUpdateOperations(d),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", *apiKey.Name)
+	d.Set("description", *apiKey.Description)
+	d.Set("enabled", *apiKey.Enabled)
+
+	// FIXME update stages from remote. api id is missing, tho, so it might be hard
```

Comment:
Can you be more specific about what needs to be done here? Maybe even link to docs or something like that would help?

I'm guessing you're talking about `stageKeys`? http://docs.aws.amazon.com/apigateway/api-reference/resource/api-key/